Expo EAS Build failing for production (iOS)

After installing react-native-flipper-performance-plugin as a devDependency, all my EAS iOS Productions builds are failing.

Command: eas build -p ios --proflie production

Log:

```
Compiling react-native-flipper-performance-plugin Pods/react-native-flipper-performance-plugin » FlipperPerformancePlugin.m

❌  (node_modules/react-native-flipper-performance-plugin/ios/FlipperPerformancePlugin.h:3:9)

  1 | #import <Foundation/Foundation.h>
  2 | 
> 3 | #import <FlipperKit/FlipperPlugin.h>
    |         ^ 'FlipperKit/FlipperPlugin.h' file not found
  4 | #import <React/RCTBridge.h>
  5 | 
  6 | @interface FlipperPerformancePlugin : NSObject<RCTBridgeModule, FlipperPlugin>
▸ ** ARCHIVE FAILED **
▸ The following build commands failed:
▸ 	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler (in target 'sovran-react-native' from project 'Pods')
▸ 	CompileC /Users/expo/Library/Developer/Xcode/DerivedData/Curebase-hlhkbcpnqfhmkyadpnbibuogbjfb/Build/Intermediates.noindex/ArchiveIntermediates/Curebase/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/react-native-flipper-performance-plugin.build/Objects-normal/arm64/FlipperPerformancePlugin.o /Users/expo/workingdir/build/node_modules/react-native-flipper-performance-plugin/ios/FlipperPerformancePlugin.m normal arm64 objective-c com.apple.compilers.llvm.clang.1_0.compiler (in target 'react-native-flipper-performance-plugin' from project 'Pods')
▸ (2 failures)
20

```

I do have FLIPPER_DISABLE on my eas.json:

```
    "production": {
      "env": {
        "APP_ENVIRONMENT": "production",
        "FLIPPER_DISABLE": "1"
      }
    }
```

Any thoughts?

Sure @Almouro .

After running eas build -p ios --profile production --no-interactive for this branch I got this:

```
    Error: Cannot find module './plugin/build'
    Require stack:
    - /Users/erisvaldojunior/Developer/mobile-app/node_modules/react-native-flipper-performance-plugin/app.plugin.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/node_modules/@expo/config-plugins/build/utils/plugin-resolver.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/node_modules/@expo/config-plugins/build/plugins/withStaticPlugin.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/node_modules/@expo/config-plugins/build/plugins/withPlugins.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/node_modules/@expo/config-plugins/build/index.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/build/project/ios/bundleIdentifier.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/build/credentials/ios/api/GraphqlClient.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/build/credentials/context.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/build/submit/context.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/build/build/runBuildAndSubmit.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/build/commands/build/index.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/node_modules/@oclif/core/lib/module-loader.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/node_modules/@oclif/core/lib/config/plugin.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/node_modules/@oclif/core/lib/config/config.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/node_modules/@oclif/core/lib/config/index.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/node_modules/@oclif/core/lib/command.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/node_modules/@oclif/core/lib/index.js
    - /Users/erisvaldojunior/.config/yarn/global/node_modules/eas-cli/bin/run
    Code: MODULE_NOT_FOUND

```

--- a/flipper-native/ios/FlipperPerformancePlugin.h
+++ b/flipper-native/ios/FlipperPerformancePlugin.h
@@ -1,8 +1,15 @@
 #import <Foundation/Foundation.h>
 
+#if __has_include(<FlipperKit/FlipperPlugin.h>)
 #import <FlipperKit/FlipperPlugin.h>
 #import <React/RCTBridge.h>
 
 @interface FlipperPerformancePlugin : NSObject<RCTBridgeModule, FlipperPlugin>
 
 @end
+
+#else
+@interface FlipperPerformancePlugin : NSObject
+
+@end
+#endif

--- a/flipper-native/ios/FlipperPerformancePlugin.m
+++ b/flipper-native/ios/FlipperPerformancePlugin.m
@@ -1,4 +1,6 @@
 #import "FlipperPerformancePlugin.h"
+
+#if __has_include(<FlipperKit/FlipperPlugin.h>)
 #import <FlipperKit/FlipperConnection.h>
 // This ensures we can use [_bridge dispatchBlock]
 #import <React/RCTBridge+Private.h>
@@ -170,3 +172,11 @@ RCT_REMAP_METHOD(killUIThread,
 }
 
 @end
+
+#else
+
+@implementation FlipperPerformancePlugin
+
+@end
+
+#endif

